### DISCUSSION TITLE
Please help! I have 'ValueError: No samples to concatenate' error while using the trainer.test functionality

### DISCUSSION THREAD

**Discussion by 1amrutesh** [open-edge-platform/anomalib#701] [Q&A] [answered]
I want to use the anomalib library on a custom dataset. However, even after I specify the path, I get the runtimeerror. The structure of the directory is 
- Bad
- - -Images
- Good_train
- --Images
- Good_test
- --Images
The images are in the Images folder. There are 200 abnormal images and 1800 normal images with 80:20 split in good_train and good_test respectively. I used the Padim model.
I have tried to implement this on Kaggle with P100 GPU. I carefully followed the [kaggle guide]

I have also attached the screenshot of the error. I shall be obliged if anyone helps me.  
![value_error](https://user-images.githubusercontent.com/94977077/201133783-e67be70b-c0dc-4a43-b39d-6632f5e6b885.png)

**Comment by djdameln** [comment, marked as answer]
Hi, it's hard to say for sure without knowing the contents of your `config.yaml`, but the most likely explanation is that the task type for your dataset is configured as `segmentation`, while your dataset does not have ground truth pixel-level annotations available. As explained in [this answer](https://github.com/openvinotoolkit/anomalib/discussions/751#discussioncomment-4281889), you will have to set the `dataset.task` parameter in the `config.yaml` to `classification` in this case.

**Reply by 1amrutesh** [reply]
Yes, thanks a lot for your help. After setting the task 'classification' instead of 'segmentation', I can run the code `trainer.test(model=model, datamodule=datamodule)` successfully, and the above error does not occur. However, the classification task can only classify whether the image is anomalous; it cannot localize it. Is it possible to localize the anomalies in the image without the 'pixel-level annotations'?

**Reply by djdameln** [reply]
Yes, after training the model with the classification task type, you can use the model to generate pixel-level predictions during inference. Just create a new `config.yaml` in which you set the `task` parameter to `segmentation` and run the [inference entrypoint script](https://github.com/openvinotoolkit/anomalib/blob/main/tools/inference/lightning_inference.py).

Please be aware that Anomalib uses the validation set to adaptively compute an optimal value for the anomaly score threshold on both the image- and pixel-level. In the absence of pixel-level annotations, the image-level threshold will be re-used as pixel-level threshold, which may negatively impact the quality of the segmentation predictions.
